backend/src/schema/resolvers/searches/queryString.spec.js

```diff
@@ -0,0 +1,43 @@
+import { queryString, escapeSpecialCharacters, normalizeWhitespace } from './queryString'
+
+describe('queryString', () => {
+  describe('special characters', () => {
+    it('does escaping correctly', () => {
+      expect(escapeSpecialCharacters('+ - && || ! ( ) { } [ ] ^ " ~ * ? : \\ / ')).toEqual(
+        '\\+ \\- \\&\\& \\|\\| \\! \\( \\) \\{ \\} \\[ \\] \\^ \\" \\~ \\* \\? \\: \\\\ \\/ ',
+      )
+    })
+  })
+
+  describe('whitespace', () => {
+    it('is normalized correctly', () => {
+      expect(normalizeWhitespace(' a \t \n b \n   ')).toEqual('a b')
+    })
+  })
+
+  describe('exact match', () => {
+    it('boosts score by factor 8', () => {
+      expect(queryString('a couple of words')).toContain('"a couple of words"^8')
+    })
+  })
+
+  describe('match all words exactly', () => {
+    it('boosts score by factor 4', () => {
+      expect(queryString('a couple of words')).toContain(
+        '("a" AND "couple" AND "of" AND "words")^4',
+      )
+    })
+  })
+
+  describe('match at least one word exactly', () => {
+    it('boosts score by factor 2', () => {
+      expect(queryString('a couple of words')).toContain('"a"^2 "couple"^2 "of"^2 "words"^2')
+    })
+  })
```

Comment:
I guess these tests give a hint of how the boosting works... I would infer that 8 is the maximum and you can evenly distribute that amongst words?

Reply:
As far as I know, there is no maximum.